processor/tailsamplingprocessor/sampling/string_tag_filter.go

```diff
@@ -15,31 +15,64 @@
 package sampling
 
 import (
+	"regexp"
+
+	"github.com/golang/groupcache/lru"
 	"go.opentelemetry.io/collector/consumer/pdata"
 	"go.uber.org/zap"
 )
 
+const defaultCacheSize = 128
+
 type stringAttributeFilter struct {
-	key    string
-	values map[string]struct{}
-	logger *zap.Logger
+	key             string
+	values          map[string]struct{}
+	regexStrEnabled bool
+	regexStrSetting *regexStrSetting
+	logger          *zap.Logger
+}
+
+type regexStrSetting struct {
+	matchedAttrs *lru.Cache
+	filterList   []*regexp.Regexp
 }
 
 var _ PolicyEvaluator = (*stringAttributeFilter)(nil)
 
 // NewStringAttributeFilter creates a policy evaluator that samples all traces with
 // the given attribute in the given numeric range.
-func NewStringAttributeFilter(logger *zap.Logger, key string, values []string) PolicyEvaluator {
+func NewStringAttributeFilter(logger *zap.Logger, key string, values []string, regexMatchEnabled bool, evictSize int) PolicyEvaluator {
+	// initialize regex filter rules and LRU cache for matched results
+	if regexMatchEnabled {
+		if evictSize == 0 {
```

Comment:
In validate config, remember to check the evictSize is >= 0, though if you give it `-1` `lru.New` won't panic, the behaviour should be it never caches anything because size > c.Max always hold

https://github.com/golang/groupcache/blob/41bb18bfe9da5321badc438f91158cd790a33aa3/lru/lru.go#L68

Reply:
yep, changed it to `if evictSize <= 0` to cover the edge case for now.